### PR TITLE
[HttpKernel] some tweaks to the controller argument resolver

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |
- update some docblocks
- remove the `LegacyArgumentResolver` class
- simplify the `TraceableControllerResolver`

### CODE REVIEW

**Comment by iltar** [comment]
:+1: You're right that this is probably better due to the introduction of a legacy class.

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolverInterface.php [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * An ArgumentResolverInterface instance knows how to determine the
  * arguments for a specific action.
  *
- * @author Fabien Potencier <fabien@symfony.com>
+ * @author Iltar van der Berg <kjarli@gmail.com>
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolverInterface.php]: I don't care too much, but this should be reverted. The code is exactly the same as the one from the original interface.

Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolverInterface.php]: No, you are right. I reverted this change and the one in the `TraceableArgumentResolver`.

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**